### PR TITLE
Use explicit NIO imports

### PR DESCRIPTION
**Pull request by fabianfett**
In NIO `2.32.0` the NIO module was split up into smaller submodules. This pr updates all imports to import only the needed modules.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -33,29 +33,32 @@ let package = Package(
...
                 .product(name: "NIOHTTP2", package: "swift-nio-http2"),
                 .product(name: "NIOSSL", package: "swift-nio-ssl"),
-                .product(name: "NIOConcurrencyHelpers", package: "swift-nio"),
                 .product(name: "NIOHTTPCompression", package: "swift-nio-extras"),
-                .product(name: "NIOFoundationCompat", package: "swift-nio"),
+                .product(name: "NIOSOCKS", package: "swift-nio-extras"),
                 .product(name: "NIOTransportServices", package: "swift-nio-transport-services"),
                 .product(name: "Logging", package: "swift-log"),
-                .product(name: "NIOSOCKS", package: "swift-nio-extras"),
             ]
```
Comment (fabianfett) [inline Package.swift:47]: Just reorderd, such that modules belonging to the same dependency are next to each other.

File: Package.swift [hunk 2]
```diff
@@ -33,29 +33,32 @@ let package = Package(
...
-                .product(name: "NIOHTTP2", package: "swift-nio-http2"),
-                "AsyncHTTPClient",
+                .product(name: "NIOEmbedded", package: "swift-nio"),
                 .product(name: "NIOFoundationCompat", package: "swift-nio"),
                 .product(name: "NIOTestUtils", package: "swift-nio"),
-                .product(name: "Logging", package: "swift-log"),
+                .product(name: "NIOSSL", package: "swift-nio-ssl"),
+                .product(name: "NIOHTTP2", package: "swift-nio-http2"),
                 .product(name: "NIOSOCKS", package: "swift-nio-extras"),
+                .product(name: "Logging", package: "swift-log"),
```
Comment (fabianfett) [inline Package.swift:61]: Again. Just reordered.

File: Package.swift [hunk 3]
```diff
@@ -33,29 +33,32 @@ let package = Package(
...
                 .product(name: "NIOTransportServices", package: "swift-nio-transport-services"),
                 .product(name: "Logging", package: "swift-log"),
-                .product(name: "NIOSOCKS", package: "swift-nio-extras"),
             ]
         ),
         .testTarget(
             name: "AsyncHTTPClientTests",
             dependencies: [
-                .product(name: "NIO", package: "swift-nio"),
+                .byName(name: "AsyncHTTPClient"),
```
Comment (glbrntt) [inline Package.swift:52]: nit: `target(name:)` is more specific

**Review by glbrntt — COMMENTED** [review]
Aside from the unhappy CI this looks fine.

**Pull request merged**